azurerm/resource_arm_action_group.go

```diff
@@ -0,0 +1,313 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/monitor/mgmt/2018-03-01/insights"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmActionGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmActionGroupCreateOrUpdate,
+		Read:   resourceArmActionGroupRead,
+		Update: resourceArmActionGroupCreateOrUpdate,
+		Delete: resourceArmActionGroupDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"short_name": {
+				Type:         schema.TypeString,
+				Required:     true,
```

Comment:
is this only required for the SMS Receiver? if so - should this be optional?

Reply:
It is only used in Email and SMS, but is required for the whole resource. Otherwise Azure complains with `"Code": "GroupShortNameIsNullOrEmpty"`.

---
In reply to: [207882329](https://github.com/terraform-providers/terraform-provider-azurerm/pull/1725#discussion_r207882329) [](ancestors = 207882329)